code/game/objects/items/cards_ids.dm

```diff
@@ -871,6 +923,54 @@
 		return insert_money(interacting_with, user) ? ITEM_INTERACT_SUCCESS : ITEM_INTERACT_BLOCKING
 	return NONE
 
+/obj/item/card/id/item_ctrl_click(mob/user)
+	if(!in_contents_of(user) || user.incapacitated) //Check if the ID is in the ID slot, so it can be changed from there too.
+		return
+
+	if(!trim)
+		balloon_alert(user, "card has no trim!")
+		return
+
+	if(!length(trim.honorifics))
+		balloon_alert(user, "card has no honorific to use!")
+		return
+
+	var/list/choices = list()
+	var/list/readable_names = HONORIFIC_POSITION_BITFIELDS()
+	for(var/i in readable_names) //Filter out the options you don't have on your ID.
+		if(trim.honorific_positions & readable_names[i]) //If the positions list has the same bit value as the readable list.
+			choices += i
+
+	var/chosen_position = tgui_input_list(user, "What position do you want your honorific in?", "Flair!", choices)
+	if(user.incapacitated || !in_contents_of(user))
+		return
+	var/honorific_position_to_use = readable_names[chosen_position]
+
+	honorific_position = initial(honorific_position) //In case you want to force an honorific on an ID, set a default that won't always be NONE.
+	honorific_title = null //We reset this regardless so that we don't stack titles on accident.
+
+	if(honorific_position_to_use & HONORIFIC_POSITION_NONE)
+		balloon_alert(user, "honorific disabled")
+	else
+		chosen_honorific = tgui_input_list(user, "What honorific do you want to use?", "Flair!!!", trim.honorifics)
+		if(!chosen_honorific || user.incapacitated || !in_contents_of(user))
+			return
+		switch(honorific_position_to_use)
```

Comment:
This has to be done otherwise the sanity check is pointless (as it updates the var regardless of if the mob can) 
```suggestion
		var/new_honorific = tgui_input_list(user, "What honorific do you want to use?", "Flair!!!", trim.honorifics)
		if(!new_honorific || user.incapacitated || !in_contents_of(user))
			return
		chosen_honorific = new_honorific
		switch(honorific_position_to_use)
```